bpmn-model/src/main/java/io/camunda/zeebe/model/bpmn/validation/zeebe/EscalationValidator.java

```diff
@@ -0,0 +1,149 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.model.bpmn.validation.zeebe;
+
+import io.camunda.zeebe.model.bpmn.instance.EndEvent;
+import io.camunda.zeebe.model.bpmn.instance.Escalation;
+import io.camunda.zeebe.model.bpmn.instance.EscalationEventDefinition;
+import io.camunda.zeebe.model.bpmn.instance.Process;
+import io.camunda.zeebe.model.bpmn.instance.StartEvent;
+import io.camunda.zeebe.model.bpmn.instance.SubProcess;
+import io.camunda.zeebe.model.bpmn.instance.ThrowEvent;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.camunda.bpm.model.xml.impl.ModelInstanceImpl;
+import org.camunda.bpm.model.xml.impl.util.ModelUtil;
+import org.camunda.bpm.model.xml.instance.DomElement;
+import org.camunda.bpm.model.xml.instance.ModelElementInstance;
+import org.camunda.bpm.model.xml.validation.ModelElementValidator;
+import org.camunda.bpm.model.xml.validation.ValidationResultCollector;
+
+public class EscalationValidator implements ModelElementValidator<Escalation> {
+
+  @Override
+  public Class<Escalation> getElementType() {
+    return Escalation.class;
+  }
+
+  @Override
+  public void validate(
+      final Escalation element, final ValidationResultCollector validationResultCollector) {
+    if (isReferredByThrowEvent(element) || isReferredByEndEvent(element)) {
+      validateEscalationCode(element, validationResultCollector);
+    } else if (isReferredByEventSubProcessStartEvent(element)) {
+      validateIfReferredByEventSubProcessStartEvent(element, validationResultCollector);
+    }
+  }
+
+  private void validateEscalationCode(
+      final Escalation element, final ValidationResultCollector validationResultCollector) {
+    if (element.getEscalationCode() == null || element.getEscalationCode().isEmpty()) {
+      validationResultCollector.addError(0, "EscalationCode must be present and not empty");
+    }
+  }
+
+  private void validateIfReferredByEventSubProcessStartEvent(
+      final Escalation element, final ValidationResultCollector validationResultCollector) {
+    final Collection<StartEvent> startEvents =
+        element.getParentElement().getChildElementsByType(SubProcess.class).stream()
+            .flatMap(p -> p.getChildElementsByType(StartEvent.class).stream())
+            .collect(Collectors.toList());
+    final long numReferredStartEvents =
+        startEvents.stream()
+            .flatMap(i -> i.getEventDefinitions().stream())
+            .filter(
+                e ->
+                    e instanceof EscalationEventDefinition
+                        && ((EscalationEventDefinition) e).getEscalation() == element)
+            .count();
+
+    if (numReferredStartEvents > 1) {
```

Comment:
❌ This is basically the same check as in the `isReferredByEventSubProcessStartEvent` method. I think we could reduce the duplication by combining the check and the validation.